### DISCUSSION TITLE
Ladder resistors network on LyliGo Tdisplay

### DISCUSSION THREAD

**Discussion by appazbmi** [alf45tar/PedalinoMini#509] [Q&A]
Hi,
First, Congrats for this great idea and dev !  
I am trying to use and set my Pedalinomini with 6 switches ladder mode + 2 switches momentary mode on a LyliGo Tdisplay (Last firmware) but the ladder settings won't work.
Can you confirm this :
Ladder > Pin_A4 = pin nr 17 on board pullup resistor =10k on 3,3v
Momentary > Pin_D1= pin nr 5 and Pin_D2 = pin nr 3
The voltage (mV) depending on switch gives with resistors on pin A4
0
637
1228
1734
2264
3020

so setting would be :
0
198
383
540
705
941

The automatic ladder on boot gives 80 80 80 ...1023
Strange ?
Thanks for your help.

**Comment by alf45tar** [comment]
D1 on 5 and D2 on 3 is valid only for T-Display S3.
Which board are you using? T-Display or T-Display S3?

**Comment by appazbmi** [comment]
T-Display S3.

But firstable what about the ladder resistor setting ? Is there any trick or specific attention ?
This is like something around electronic is wrong ? The values voltage on scope is OK on pin A4 ? 
My cabling 
![Capture d’écran du 2024-04-28 22-11-35](https://github.com/alf45tar/PedalinoMini/assets/16247436/afbce528-89cc-49fe-b463-7f5e53a6dfb4)
![Capture d’écran du 2024-04-28 22-16-44](https://github.com/alf45tar/PedalinoMini/assets/16247436/f78d5b2c-2947-421b-8b21-623515ab4970)

**Comment by alf45tar** [comment]
It looks ok but which version of ladder are you using?

**Comment by appazbmi** [comment]
The 1st one with 2k,3k 5k....
Is there any possibility to read the dynamic adc Pin_A4 value ?
By the way my main settings : 
![Capture d’écran du 2024-04-29 11-18-01](https://github.com/alf45tar/PedalinoMini/assets/16247436/2e5ebff0-1058-49ca-9283-2a1a4b650c5f)
![Capture d’écran du 2024-04-29 11-18-46](https://github.com/alf45tar/PedalinoMini/assets/16247436/9bcb9735-b45b-4a2f-89b9-b5b152690b0c)
![Capture d’écran du 2024-04-29 11-19-28](https://github.com/alf45tar/PedalinoMini/assets/16247436/d2f6cc2d-fae7-4774-befe-5a2b1208f45f)
![Capture d’écran du 2024-04-29 11-20-18](https://github.com/alf45tar/PedalinoMini/assets/16247436/c21b5ffa-fe19-4f5b-ba46-ef634b2eea59)
Thanx

**Comment by alf45tar** [comment]
Only Pedal 4 must be set as Ladder.

**Reply by appazbmi** [reply]
OK thanks, that was just a question of web setting :) 
Now lets start wiring & use the led WS2812B.

Thanks again !
